synapse/tests/test_lib_config.py

```diff
@@ -0,0 +1,44 @@
+import os
+
+import synapse.exc as s_exc
+import synapse.common as s_common
+
+import synapse.tests.utils as s_test
+
+import synapse.lib.config as s_config
+
+class ConfTest(s_test.SynTest):
+
+    async def test_config_base(self):
+
+        confdefs = (
+            ('foo', 20, int),
+        )
+
+        conf = s_config.Config(confdefs)
+
+        with self.raises(s_exc.NoSuchName):
+            await conf.set('hehe', 'haha')
+
+        await conf.loadConfDict({'foo': 30})
+
+        self.eq(conf.get('foo'), 30)
+
+        with self.getTestDir() as dirn:
+
+            path = os.path.join(dirn, 'foo.yaml')
+            with s_common.genfile(path) as fd:
+                fd.write(b'foo: 8080')
+
+            await conf.loadConfYaml(path)
+
+        self.eq(conf.get('foo'), 8080)
+
+        os.environ['SYN_CONF_TEST_FOO'] = '31337'
```

Comment:
Use `self.setTstEnvars` as a context manager here so that we're not munging os.environ for other tests.